### PR TITLE
User store migration docs

### PR DESCRIPTION
**Pull request by JKAUSHALYA**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,8 +12,11 @@ To run the project on local, it requires [python](https://www.python.org/downloa
 
 ## Run project locally (Dev Mode)
 
-1. Clone repo
-2. Run `mkdocs serve` command in `<Lang folder>`. E.g. `cd en && mkdocs serve`
+1. Clone the repo
```
Comment (gomathyk) [inline README.md:15]: Shall we remove the "Then"s and add full stops?

File: en/docs/setup/migrating-userstore-managers.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
+# Migrating User Stores
+
+Before doing the actual user store migration, it is recommended to do a dry run and analyze the generated report for 
+any recommendations related to the migration. 
+
+## Dry Run
+
+Dry-run capability in the user store migrator will allow the migration utility to validate the system for the 
+user store configurations and generate a report regarding the migration compatibility. In the generated migration report, 
+if there are any warnings it is recommended to contact the WSO2 support to identify the best migration strategy.
```
Comment (gomathyk) [inline en/docs/setup/migrating-userstore-managers.md:10]: If there are any warnings in the generated migration report, ...

File: en/docs/setup/migrating-userstore-managers.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+### How to Run
+
+Configure the migration report path using the value of “reportPath” in the migration-config.yaml.
```
Comment (gomathyk) [inline en/docs/setup/migrating-userstore-managers.md:14]: ...using the value of `reportPath` in `<IS_HOME>/migration-resources/migration-config.yaml`.

File: en/docs/setup/migrating-userstore-managers.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+### How to Run
+
+Configure the migration report path using the value of “reportPath” in the migration-config.yaml. 
+This path should be an absolute path. 
+
+Run the migration utility with system property “dryRun”. 
+
+Ex:
+
+Unix/Linux
```
Comment (gomathyk) [inline en/docs/setup/migrating-userstore-managers.md:21]: Shall we give this in code tabs? 

``` tab="Unix/Linux"
sh wso2server.sh -Dmigrate -Dcomponent=identity -DdryRun
```
``` tab="Windows"
wso2server.bat -Dmigrate -Dcomponent=identity -DdryRun
```

File: en/docs/setup/migrating-userstore-managers.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+``` bash
+wso2server.bat -Dmigrate -Dcomponent=identity -DdryRun
+```
+ 
+After that analyze the generated report that resides in the provided location.
+
+## Migrating User Stores
+
+If you have multiple tenants and multiple user stores and you just only need to migrate a few of them, refer to the 
+configuration section below on how to migrate few of them. If you need to migrate all of them, use the “migrateAll”
```
Comment (gomathyk) [inline en/docs/setup/migrating-userstore-managers.md:38]: Remove "just" in 
...and you only need to migrate a few of the...

Remove "on how to migrate few of them"

use the `migrateAll`

File: en/docs/setup/migrating-userstore-managers.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+## Migrating User Stores
+
+If you have multiple tenants and multiple user stores and you just only need to migrate a few of them, refer to the 
+configuration section below on how to migrate few of them. If you need to migrate all of them, use the “migrateAll” 
+property (This is set to true by default).  
+
+### Special Scenarios
+
+If the user store is an LDAP user store and SCIM enabled for that user store, migrating that user store is not required.
```
Comment (gomathyk) [inline en/docs/setup/migrating-userstore-managers.md:43]: and SCIM is enabled

File: en/docs/setup/migrating-userstore-managers.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+If you have multiple tenants and multiple user stores and you just only need to migrate a few of them, refer to the 
+configuration section below on how to migrate few of them. If you need to migrate all of them, use the “migrateAll” 
+property (This is set to true by default).  
+
+### Special Scenarios
+
+If the user store is an LDAP user store and SCIM enabled for that user store, migrating that user store is not required. 
+As SCIM will create a user id for the users in that user store, SCIM id can be used as the unique user id. To do that, 
+change the “user_id_attribute” to the value of SCIM id, in deployment toml.
```
Comment (gomathyk) [inline en/docs/setup/migrating-userstore-managers.md:45]: `user_id_attribute`

in the `<IS_HOME>/repository/conf/deployment.toml` file.

File: en/docs/setup/migrating-userstore-managers.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+
+## Configurations
+
+| Property Name | Description |
+| ------------- | ----------- |
+| migrateAll | Migrate all the tenants and all the user store domains in it. |
+| reportPath | Absolute path for the dry report. This is required in the dry run mode. |
+
+
+Following configurations are only needed whenever migrating few tenants. Configuration format should be like below. tenant-param is a place holder name to represent each tenant uniquely. Ex: tenant1
```
Comment (gomathyk) [inline en/docs/setup/migrating-userstore-managers.md:55]: needed to migrate a few few tenants. Configuration format should be similar to the one mentioned below.

File: en/docs/setup/migrating-what-has-changed.md [hunk 9]
```diff
@@ -375,4 +375,13 @@ SMS provider.
     ```toml
     [authentication.authenticator.sms_otp.parameters]
     UseInternalErrorCodes = false
-    ```
\ No newline at end of file
+    ```
+## New User Store Managers with built in unique id support. 
+
+New user store managers with inbuilt unique id support will provide the capability to enhance the user store operations 
+that were initially architecturally impossible. Such as username rename functionality and significant performance
```
Comment (gomathyk) [inline en/docs/setup/migrating-what-has-changed.md:382]: Shall we rephrase this? 

These operations include **username rename** functionality and a **significant performance increase**.

File: en/docs/setup/migrating-what-has-changed.md [hunk 10]
```diff
@@ -375,4 +375,13 @@ SMS provider.
...
     UseInternalErrorCodes = false
-    ```
\ No newline at end of file
+    ```
+## New User Store Managers with built in unique id support. 
+
+New user store managers with inbuilt unique id support will provide the capability to enhance the user store operations 
+that were initially architecturally impossible. Such as username rename functionality and significant performance 
+increase as well. So migrating to new user store managers will allow us to use those new features whenever those are 
+available.
```
Comment (gomathyk) [inline en/docs/setup/migrating-what-has-changed.md:384]: those new features available.

File: en/docs/setup/migrating-what-has-changed.md [hunk 11]
```diff
@@ -375,4 +375,13 @@ SMS provider.
...
\ No newline at end of file
+    ```
+## New User Store Managers with built in unique id support. 
+
+New user store managers with inbuilt unique id support will provide the capability to enhance the user store operations 
+that were initially architecturally impossible. Such as username rename functionality and significant performance 
+increase as well. So migrating to new user store managers will allow us to use those new features whenever those are 
+available.  
+
+New user store managers are named with the “UniqueiId” prefix for the old user store managers. For example
```
Comment (gomathyk) [inline en/docs/setup/migrating-what-has-changed.md:386]: `UniqueiId`

For example,

File: en/docs/setup/migrating-to-5100.md [hunk 12]
```diff
@@ -188,8 +188,11 @@
         If the user is not already a superuser, assign the permission before starting the migration.
             
             ALTER USER <user> WITH SUPERUSER;
+
+10. If you are migrating your user stores to the new user store managers with the unique id capabilities, please follow
```
Comment (sherenewso2) [inline en/docs/setup/migrating-to-5100.md:192]: id --> ID 

avoid using "please" in docs to maintain a neutral tone

File: en/docs/setup/migrating-userstore-managers.md (hunk 4 above)
Comment (sherenewso2) [reply en/docs/setup/migrating-userstore-managers.md:21]: syntax for code tabs:

```
    ```bash tab="Unix/Linux
     enter code here
    ```
```

File: en/docs/setup/migrating-what-has-changed.md [hunk 13]
```diff
@@ -375,4 +375,13 @@ SMS provider.
     ```toml
     [authentication.authenticator.sms_otp.parameters]
     UseInternalErrorCodes = false
-    ```
\ No newline at end of file
+    ```
+## New User Store Managers with built in unique id support.
```
Comment (sherenewso2) [inline en/docs/setup/migrating-what-has-changed.md:379]: id --> ID

Comment (sherenewso2) [reply en/docs/setup/migrating-what-has-changed.md:379]: please apply to all instances

**Comment by JKAUSHALYA** [comment]
Fixed requested changes.

**Pull request merged**